src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidUnreliableStreamRead.cs

```diff
@@ -130,6 +141,11 @@ public bool IsAnyStreamReadAsyncMethod(IMethodSymbol method)
                     SymbolEqualityComparer.Default.Equals(method, m) || IsOverrideOf(method, m));
             }
 
+            public bool IsKnownReliableStreamType(ITypeSymbol? type)
+            {
+                return _knownReliableStreamTypes.Any(t => SymbolEqualityComparer.Default.Equals(type, t));
```

Comment:
Can the set be created with this comparer so the lookup is O(1) rather than O(N)?